### ISSUE TITLE
typo in example loss

### ISSUE THREAD

**Issue by mthorrell** [KindXiaoming/grow-crystals#6]
the toy example has this code:
```python
prob = distlayer(inputs)
prob = prob/torch.sum(prob, dim=1, keepdim=True)
logits = torch.log(prob)
loss = torch.nn.functional.cross_entropy(logits, labels)
```

but since these are log-probabilities the last line should be `-logits[range(labels.size(0)), labels].mean()`

also rename logits to log_probs?

**Comment by KindXiaoming** [comment]
Sorry I don't think I understand your suggestion. The second-to-last line is to turn probabilities back to standard logits (as used in cross-entropy loss), and then we can simply call the cross-entropy loss as one would normally call it (taking in logits and labels).

**Comment by mthorrell** [comment]
Answering this turned out more difficult than I initially thought.  Numerically, it is true that cross_entropy(log(p)) == log loss(p). BUT, I would say that, in general, logits (and how they are used) are usually expected to vary between -inf and inf. log(p) only varies between -inf and 0.

This doesn't numerically matter for cross-entropy but this will make a difference in, for example, binary cross-entropy. There, logit(p) = log(p / (1-p)) which of course is not equal to log(p).

It may also be worth mentioning that in other code here (e.g. mnist.ipynb), `outputs[range(targets.size(0)), targets].mean()` is used where outputs is -log(prob). This construction is a more explicitly constructed log-loss which seems preferable.

**Comment by hichoe95** [comment]
Since PyTorch’s cross-entropy loss function internally applies the log softmax operation, you don’t need to do it again manually. (https://github.com/KindXiaoming/grow-crystals/blob/60a67221936c860508aec06b01250ca80799b326/src/utils/model.py#L267)

Therefore, after running the forward pass of your MLP_HS class, computing the loss as 
`-losses = -logits[range(labels.size(0)), labels].mean()` is a correct approach. This directly selects the log probabilities for the true classes and computes their negative mean, which is equivalent to what F.cross_entropy would compute internally.
